interface/src/error.rs

```diff
@@ -0,0 +1,122 @@
+use num_traits::{FromPrimitive, ToPrimitive};
+use serde::Serialize;
+use solana_decode_error::DecodeError;
+
+// Use strum when testing to ensure our FromPrimitive
+// impl is exhaustive
+#[cfg_attr(test, derive(strum_macros::FromRepr, strum_macros::EnumIter))]
```

Comment:
We used strum in tests to ensure that the conversion code was exhaustive -- can you add a test similar to https://github.com/anza-xyz/agave/blob/ccbe01dd332ab6b75cd2091fb359e4ad8fea0c2c/sdk/pubkey/src/lib.rs#L1372 to check that?